miden-tx/src/compiler/mod.rs

```diff
@@ -440,6 +440,7 @@ fn recursively_collect_call_branches(code_block: &CodeBlock, branches: &mut Vec<
         }
         CodeBlock::Span(_) => {}
         CodeBlock::Proxy(_) => {}
+        CodeBlock::Dyn(_) => {}
```

Comment:
This bugfix is blocking the other PRs. I'm going to merge this PR to unblock the others.